### ISSUE TITLE
go remove unused imports doesn't work (golang layer)

### ISSUE THREAD

**Issue by maryjmc** [syl20bnr/spacemacs#5855]
When I try to insert an unused package then run `go-remove-unused-import` (`SPC m i r`) it says removes 0 packages even compilers raise unused package error.

Reproduction guide

* Start emacs
* Open `main.go ` file with `SPC f f`
* Run `go-remove-unused-imports` with `SPC m i r`
* Get result `Removed 0 imports` (even when I run this file, the compiler raise error unused imports `fmt`)

The `main.go` file look like this

```
package main

import (
        "fmt"
	"log"
	"os"
	_ "github.com/goinaction/code/chapter2/sample/matchers"
	"github.com/goinaction/code/chapter2/sample/search"
)
func init() {
	log.SetOutput(os.Stdout)
}
func main() {
	search.Run("president")
```

**Comment by NJBS** [comment]
Could you update your issue to reflect [this template](https://github.com/syl20bnr/spacemacs/wiki/Debugging#how-to-make-a-great-bug-report) such that others can better assist you?

Additionally, a [short example](http://sscce.org/) demonistrating how to reproduce your issue would be great! :smile:

**Comment by maryjmc** [comment]
I have updated my issues as you suggested. Also I have looked this issues [https://github.com/dominikh/go-mode.el/issues/19](url) before but the author said that he have fixed it and it worked. So I think maybe it has trouble with spacemacs, I don't sure. That why I added this issues in here.

**Comment by StreakyCobra** [comment]
Can you try to reproduce this on stock emacs with just go-mode enabled (`SPC q D` may help).

**Comment by ntcong** [comment]
go 1.5 and 1.5.1 have a regression that it doesn't support build to /dev/null

It has since fixed but the fix is in 1.6
https://github.com/golang/go/issues/12407

Can you confirm the go version you are using?

**Comment by maryjmc** [comment]
@StreakyCobra I have reproduce this on stock emacs like this

* restart emacs with just go-mode enabled `SPC q d` then type `go-mode` 
* write the sample code above
* save file
* run command with `C-x C-w` then type `go-remove-unused-imports`
* the result is the same `Removed 0 imports`

@ntcong Yes. I used go version `1.6.1`. Here is the output when I checkout my version with `go version`

```
go version go1.6.1 darwin/amd64
```

**Comment by StreakyCobra** [comment]
@maryjmc So this means that's an upstream bug if it happens without Spacemacs. You should probably report it there :smiley:

**Comment by maryjmc** [comment]
@NJBS @ntcong @StreakyCobra 

Thank you so much for all your help. I have found the solution for my trouble. First let me said this, I followed the instruction to configure `go` layer in spacemacs at [spacemacs go layer doc](https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Blang/go). And after finish the setup like the docs said, I have a trouble like above. But today, when I look back at the [Go layer pre-requisites](https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Blang/go#pre-requisites), I realise that it has nothing to do with `goimports`. So I have installed `goimports` with this command `go get golang.org/x/tools/cmd/goimports`. After that, everything works like a charm.

**Comment by NJBS** [comment]
PR for documentation update https://github.com/syl20bnr/spacemacs/pull/5873

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.

**Comment by hikkikuma1991** [comment]
I still have met this problem, and I have done `go get golang.org/x/tools/cmd/goimports`. My go version is `go version go1.10 darwin/amd64`. My spacemacs is `0.200.13@25.3.1 (spacemacs)`. Could anyone help me with this problem?
